src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -873,6 +873,28 @@ internal bool CanPaste(object parameters)
 
         private void Paste(object parameter)
         {
+            var locatableModels = model.ClipBoard.Where(item => item is NoteModel || item is NodeModel);
```

Comment:
All of the checking of bounds, and mouse positions should happen in the view, and not use properties on the model for the reasons I describe below.

I recommend raising an event here `RequestPaste`, which is handled by the view, which can then look at the actual sizes of elements, whether they are in the bounds of the view or not, and then call down to the model to actually paste with the appropriate offset.